source/rules/literals.rs

```diff
@@ -42,25 +42,26 @@ use nom::{
     hex_digit,
     oct_digit
 };
-use std::str;
 use std::str::FromStr;
+use std::str;
+use super::super::ast::Literal;
 
 named!(
-    pub null< Option<()> >,
+    pub null<Literal>,
     map_res!(
         tag!("null"),
```

Comment:
Even `NuLl` is valid :wink:.
Good catch. I will address as a new issue to not pollute this PR.